docs/concepts.rst

```diff
@@ -47,105 +58,119 @@ Here is a sample template model::
+        o String licenseeState
+        o String licenseeEntityType
+        o String licenseeAddress
 
-    /** 
-    * Execute the smart clause 
-    * @param {Context} context - the Accord context 
-    * @param {io.clause.latedeliveryandpenalty.LateDeliveryAndPenaltyRequest}
-    * context.request - the incoming request 
-    * @param {io.clause.latedeliveryandpenalty.LateDeliveryAndPenaltyResponse}
-    * context.response - the response * @AccordClauseLogic 
-    */
-    function execute(context) {
+        ...
 
-        logger.info(context); var req = context.request; var res =
-        context.response; var data = context.data; var now =
-        moment(req.timestamp); var agreed = moment(req.agreedDelivery);
+        /* payment */
+        o PaymentClauseModel paymentClause
+    }
 
-        res.buyerMayTerminate = false; res.penalty = 0;
+    asset PaymentClauseModel extends ClauseModel {
+        o String amountText
+        o Double amount
+        o String paymentProcedure
+    }
 
-        if (req.forceMajeure) {
-            logger.info('forceMajeure'); res.buyerMayTerminate = true;
-        }
+In this example ``paymentClause`` refers to a separate model definition that describes the structure of an inline payment clause, ``PaymentClauseModel``.
 
-        if (!req.forceMajeure && now.isAfter(agreed)) {
-            logger.info('late'); logger.info('penalty duration unit: ' +
-            data.penaltyDuration.unit); logger.info('penalty duration amount: '
-            + data.penaltyDuration.amount); // the delivery is late var diff =
-            now.diff(agreed, data.penaltyDuration.unit); logger.info('diff:' +
-            diff);
+This allows you to build contracts that are composed of reusable clause definitions.
 
-            var penalty = (diff / data.penaltyDuration.amount) *
-            data.penaltyPercentage/100 * req.goodsValue;
+.. note:: Clause models are ``asset``s not ``concept``s and extend the built-in ``ClauseModel`` model.
 
-            // cap the maximum penalty if (penalty > data.capPercentage/100 *
-            req.goodsValue) {
-                logger.info('capped.'); penalty = data.capPercentage/100 *
-                req.goodsValue;
-            }
+Template Logic 
+--------------
 
-            res.penalty = penalty;
+The logic for a template is written in Ergo, a domain specific language for smart legal contracts. The Ergo clauses are
+invoked by the engine when transactions are received for processing and return a response. Both the 
+incoming requests and responses are modeled types.
 
-            // can we terminate? if (diff > data.termination) {
-                logger.info('buyerMayTerminate.'); res.buyerMayTerminate = true;
+Here is a sample Ergo contract::
+
+    namespace io.clause.latedeliveryandpenalty
+
+    // Imports CTO files
+    import org.accordproject.contract.*
+    import io.clause.latedeliveryandpenalty.*
+    // Date and time library
+    import ergo.moment.*
```

Comment:
@jeromesimeon ?

Reply:
@mttrbrts It won't break. This will change in Ergo soon, but for the time being it's okay. Will have to be reviewed once imports are handled properly.